### ISSUE TITLE
Larger and/or configurable _MAX_LENGTH for unittest messages

### ISSUE THREAD

**Issue by 9e13ff31-1b24-42c8-ab55-075b0e4cca29** [python/cpython#75022]
BPO | [30839](https://bugs.python.org/issue30839)
--- | :---
Nosy | @bitdancer
Dependencies | <li>bpo-27432: Support unittest assertion truncation of repr in error messages</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2017-07-03.17:59:26.581>
labels = ['3.7', 'type-feature', 'library']
title = 'Larger and/or configurable _MAX_LENGTH for unittest messages'
updated_at = <Date 2017-07-05.20:27:37.208>
user = 'https://bugs.python.org/maarten-treewalker'
```

bugs.python.org fields:
```python
activity = <Date 2017-07-05.20:27:37.208>
actor = 'r.david.murray'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2017-07-03.17:59:26.581>
creator = 'maarten-treewalker'
dependencies = ['27432']
files = []
hgrepos = []
issue_num = 30839
keywords = []
message_count = 2.0
messages = ['297605', '297779']
nosy_count = 2.0
nosy_names = ['r.david.murray', 'maarten-treewalker']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue30839'
versions = ['Python 3.7']
```

</p></details>

**Comment by 9e13ff31-1b24-42c8-ab55-075b0e4cca29** [comment]
In a unit test for one of my projects I am comparing tree-structured objects using assertEqual(). Because there can be a few nested levels, the repr() string can exceed the \_MAX_LENGTH which is just 80 characters. Not by much, but enough to get shortened and therefore requiring extra effort to figure out where the difference is.

I think it is useful to shorten the messages at some point, to avoid accidentally spamming the log with pages of output that no-one is going to read. However, 80 characters is a length that someone debugging a failing test case can easily deal with. So I think the limit could be set an order of magnitude larger.

Making the maximum length configurable would also solve the issue, but still it would be good to increase the default maximum then, in my opinion.

The discussion of bpo-27432 mentions \_MAX_LENGTH customization, but that issue is not primarily about \_MAX_LENGTH.

**Comment by bitdancer** [comment]
Yes, setting \_MAX_LENGTH can be an independent feature request.  

The default will not be changed, however, since it is the most useful default given that assertEquals in general produces a much more useful diff output in addition to the one-line summary.  You might also/alternatively want to look in to providing your own assertEqual extension to produce a useful diff of your particular data structure.
